src/inits.jl

```diff
@@ -0,0 +1,231 @@
+        dims...;
+        init_kwargs...,
+        kwargs...)
+end
+
+function glorot_normal(; init_kwargs...)
+    return (dims...; kwargs...) -> glorot_normal(_default_rng(),
+        dims...;
+        init_kwargs...,
+        kwargs...)
+end
+"""
+    kaiming_uniform(rng::AbstractRNG, size...; gain = √2f0)
+
+Return an `Array{Float32}` of the given `size` containing random numbers drawn from a
+uniform distribution on the interval `[-x, x]`, where `x = gain * sqrt(3/fan_in)`.
+
+# References
+
+[1] He, Kaiming, et al. "Delving deep into rectifiers: Surpassing human-level performance on
+imagenet classification." _Proceedings of the IEEE international conference on computer
+vision_. 2015.
+"""
+function kaiming_uniform(rng::AbstractRNG, dims::Integer...; gain::Real=√2.0f0)
+    bound = Float32(√3.0f0 * gain / sqrt(first(_nfan(dims...))))
+    return (rand(rng, Float32, dims...) .- 0.5f0) .* 2 * bound
+end
+
+function kaiming_uniform(dims::Integer...; kwargs...)
+    return kaiming_uniform(_default_rng(), dims...; kwargs...)
+end
+
+function kaiming_uniform(rng::AbstractRNG; init_kwargs...)
+    return (rng, dims...; kwargs...) -> kaiming_uniform(rng,
+        dims...;
+        init_kwargs...,
+        kwargs...)
+end
+
+function kaiming_uniform(; init_kwargs...)
+    return (dims...; kwargs...) -> kaiming_uniform(_default_rng(),
+        dims...;
+        init_kwargs...,
+        kwargs...)
+end
+"""
+    kaiming_normal(rng::AbstractRNG, size...; gain = √2f0)
+
+Return an `Array{Float32}` of the given `size` containing random numbers taken from a normal
+distribution standard deviation `gain / sqrt(fan_in)`
+
+# References
+
+[1] He, Kaiming, et al. "Delving deep into rectifiers: Surpassing human-level performance on
+imagenet classification." _Proceedings of the IEEE international conference on computer
+vision_. 2015.
+"""
+function kaiming_normal(rng::AbstractRNG, dims::Integer...; gain::Real=√2.0f0)
+    std = Float32(gain / sqrt(first(_nfan(dims...))))
+    return randn(rng, Float32, dims...) .* std
+end
+
+function kaiming_normal(dims::Integer...; kwargs...)
+    return kaiming_normal(_default_rng(), dims...; kwargs...)
+end
+
+function kaiming_normal(rng::AbstractRNG; init_kwargs...)
+    return (rng, dims...; kwargs...) -> kaiming_normal(rng,
+        dims...;
+        init_kwargs...,
+        kwargs...)
+end
+
+function kaiming_normal(; init_kwargs...)
+    return (dims...; kwargs...) -> kaiming_normal(_default_rng(),
+        dims...;
+        init_kwargs...,
+        kwargs...)
+end
+
+"""
+    truncated_normal([rng = default_rng_value()], size...; mean = 0, std = 1, lo = -2, hi = 2)
+
+Return an `Array{Float32}` of the given `size` where each element is drawn from a truncated normal distribution.
+The numbers are distributed like `filter(x -> lo<=x<=hi, mean .+ std .* randn(100))`.
+"""
+function truncated_normal(rng::AbstractRNG, dims::Integer...; mean=0, std=1, lo=-2, hi=2)
+    norm_cdf(x) = 0.5 * (1 + erf(x / √2))
```

Comment:
We should make this a regular function rather than a closure.

Also this will cause unnecessary type promotions. `norm_cdf(x::T) where {T} = T(0.5) * (1 + erf(x / √2))`